dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -1007,6 +1016,8 @@ CLanStatus::InitializeNetTaskbarNotifications()
 
         if (nid.uFlags & NIF_ICON)
             DestroyIcon(nid.hIcon);
+        
+        SetWindowLongPtr(hwndDlg, DWLP_USER, (LONG_PTR)pContext);
```

Comment:
What is the purpose of this?
the WM_INITDIALOG already sets this?

Reply:
That's precisely the problem, pointer has to be set when this function initialize the tray icon on session opening, and not on WM_INITDIALOG which is only called when the properties dialog is opened

Reply:
Not sure how you tested that, but you might wanna re-think your method of checking this:
```
kd> .reload /user
Loading User Symbols
..........................................
kd> kp
 # ChildEBP RetAddr  
00 00deec54 77aa450a netshell!LANStatusDlg(struct HWND__ * hwndDlg = 0x001000de, unsigned int uMsg = 0x110, unsigned int wParam = 0, long lParam = 0n1487376)+0x6c [R:\src\dev\dll\shellext\netshell\lanstatusui.cpp @ 843] 
01 00deec84 77a928c1 user32!CALL_EXTERN_WNDPROC+0x1a
02 00deed68 77a959de user32!IntCallWindowProcW(int IsAnsiProc = 0n0, <function> * WndProc = 0x766d0310, struct _WND * pWnd = 0x00336798, struct HWND__ * hWnd = 0x001000de, unsigned int Msg = 0x110, unsigned int wParam = 0, long lParam = 0n1487376)+0x681 [R:\src\dev\win32ss\user\user32\windows\message.c @ 1552] 
03 00deed98 77a76a68 user32!CallWindowProcW(<function> * lpPrevWndFunc = 0x766d0310, struct HWND__ * hWnd = 0x001000de, unsigned int Msg = 0x110, unsigned int wParam = 0, long lParam = 0n1487376)+0xce [R:\src\dev\win32ss\user\user32\windows\message.c @ 1872] 
04 00deedc4 77aa450a user32!DefDlgProcW(struct HWND__ * hDlg = 0x001000de, unsigned int Msg = 0x110, unsigned int wParam = 0, long lParam = 0n1487376)+0x78 [R:\src\dev\win32ss\user\user32\windows\dialog.c @ 1792] 
05 00deedf4 77a928c1 user32!CALL_EXTERN_WNDPROC+0x1a
06 00deeed8 77a968ab user32!IntCallWindowProcW(int IsAnsiProc = 0n0, <function> * WndProc = 0x77a769f0, struct _WND * pWnd = 0x00336798, struct HWND__ * hWnd = 0x001000de, unsigned int Msg = 0x110, unsigned int wParam = 0, long lParam = 0n1487376)+0x681 [R:\src\dev\win32ss\user\user32\windows\message.c @ 1552] 
07 00deef0c 77a97d1d user32!IntCallMessageProc(struct _WND * Wnd = 0x00336798, struct HWND__ * hWnd = 0x001000de, unsigned int Msg = 0x110, unsigned int wParam = 0, long lParam = 0n1487376, int Ansi = 0n0)+0x1eb [R:\src\dev\win32ss\user\user32\windows\message.c @ 1798] 
08 00deef94 77a789ad user32!SendMessageW(struct HWND__ * Wnd = 0x001000de, unsigned int Msg = 0x110, unsigned int wParam = 0, long lParam = 0n1487376)+0x11d [R:\src\dev\win32ss\user\user32\windows\message.c @ 2403] 
09 00def2dc 77a76de2 user32!DIALOG_CreateIndirect(struct HINSTANCE__ * hInst = 0x766c0000, void * dlgTemplate = 0x76702a50, struct HWND__ * owner = 0x00000000, <function> * dlgProc = 0x766d0310, long param = 0n1487376, int unicode = 0n1, struct HWND__ ** modal_owner = 0x00000000)+0xa3d [R:\src\dev\win32ss\user\user32\windows\dialog.c @ 1023] 
0a 00def304 77a76e20 user32!CreateDialogIndirectParamAorW(struct HINSTANCE__ * hInstance = 0x766c0000, struct DLGTEMPLATE * lpTemplate = 0x76702a00, struct HWND__ * hWndParent = 0x00000000, <function> * lpDialogFunc = 0x766d0310, long lParamInit = 0n1487376, unsigned long Flags = 0)+0x42 [R:\src\dev\win32ss\user\user32\windows\dialog.c @ 1628] 
0b 00def324 77a76f15 user32!CreateDialogIndirectParamW(struct HINSTANCE__ * hInstance = 0x766c0000, struct DLGTEMPLATE * lpTemplate = 0x76702a00, struct HWND__ * hWndParent = 0x00000000, <function> * lpDialogFunc = 0x766d0310, long lParamInit = 0n1487376)+0x20 [R:\src\dev\win32ss\user\user32\windows\dialog.c @ 1661] 
0c 00def348 766cfab3 user32!CreateDialogParamW(struct HINSTANCE__ * hInstance = 0x766c0000, wchar_t * lpTemplateName = 0x00000192 "--- memory read error at address 0x00000192 ---", struct HWND__ * hWndParent = 0x00000000, <function> * lpDialogFunc = 0x766d0310, long dwInitParam = 0n1487376)+0x65 [R:\src\dev\win32ss\user\user32\windows\dialog.c @ 1703] 
0d 00def7b8 766cf0a8 netshell!CLanStatus::InitializeNetTaskbarNotifications(void)+0x4f3 [R:\src\dev\dll\shellext\netshell\lanstatusui.cpp @ 954] 
0e 00def7c0 75822899 netshell!CLanStatus::Exec(struct _GUID * pguidCmdGroup = 0x758314f8 {000214D2-0000-0000-C000-000000000046}, unsigned long nCmdID = 2, unsigned long nCmdexecopt = 0, struct tagVARIANT * pvaIn = 0x00000000, struct tagVARIANT * pvaOut = 0x00000000)+0x28 [R:\src\dev\dll\shellext\netshell\lanstatusui.cpp @ 1062] 
0f 00def7f0 7582280a stobject!CSysTray::InitNetShell(void)+0x79 [R:\src\dev\dll\shellext\stobject\csystray.cpp @ 103] 
10 00def80c 75822d47 stobject!CSysTray::InitIcons(void)+0xea [R:\src\dev\dll\shellext\stobject\csystray.cpp @ 135] 
11 00def828 75823823 stobject!CSysTray::ProcessWindowMessage(struct HWND__ * hWnd = 0x000900ca, unsigned int uMsg = 1, unsigned int wParam = 0, long lParam = 0n14613120, long * lResult = 0x00def860, unsigned long dwMsgMapID = 0)+0xf7 [R:\src\dev\dll\shellext\stobject\csystray.cpp @ 346] 
12 00def894 77aa450a stobject!ATL::CWindowImplBaseT<ATL::CWindow,ATL::CWinTraits<2252341248,392> >::WindowProc(struct HWND__ * hWnd = 0x000900ca, unsigned int uMsg = 1, unsigned int wParam = 0, long lParam = 0n14613120)+0xa3 [R:\src\dev\sdk\lib\atl\atlwin.h @ 1626] 
13 00def8c4 77a92896 user32!CALL_EXTERN_WNDPROC+0x1a
14 00def9a8 77a98550 user32!IntCallWindowProcW(int IsAnsiProc = 0n0, <function> * WndProc = 0x00ce0000, struct _WND * pWnd = 0x00335ca8, struct HWND__ * hWnd = 0x000900ca, unsigned int Msg = 1, unsigned int wParam = 0, long lParam = 0n14613120)+0x656 [R:\src\dev\win32ss\user\user32\windows\message.c @ 1547] 
15 00defa48 7c93774c user32!User32CallWindowProcFromKernel(void * Arguments = 0x00defa60, unsigned long ArgumentLength = 0x56)+0x360 [R:\src\dev\win32ss\user\user32\windows\message.c @ 3015] 
16 00defc38 77a9f496 ntdll!KiUserCallbackDispatcher+0x2c
17 00defce4 75822114 user32!CreateWindowExW(unsigned long dwExStyle = 0x188, wchar_t * lpClassName = 0x0000c06d "--- memory read error at address 0x0000c06d ---", wchar_t * lpWindowName = 0x00000000 "", unsigned long dwStyle = 0x86400000, int x = 0n-2147483648, int y = 0n-2147483648, int nWidth = 0n-2147483648, int nHeight = 0n-2147483648, struct HWND__ * hWndParent = 0x00000000, struct HMENU__ * hMenu = 0x00000000, struct HINSTANCE__ * hInstance = 0x75820000, void * lpParam = 0x00000000)+0x3e6 [R:\src\dev\win32ss\user\user32\windows\window.c @ 609] 
18 00defd24 7582201d stobject!ATL::CWindowImplBaseT<ATL::CWindow,ATL::CWinTraits<2252341248,392> >::Create(struct HWND__ * hWndParent = 0x00000000, class ATL::_U_RECT rect = class ATL::_U_RECT, wchar_t * szWindowName = 0x00000000 "", unsigned long dwStyle = 0x86400000, unsigned long dwExStyle = 0x188, class ATL::_U_MENUorID MenuOrID = class ATL::_U_MENUorID, unsigned short atom = 0xc06d, void * lpCreateParam = 0x00000000)+0xd4 [R:\src\dev\sdk\lib\atl\atlwin.h @ 1676] 
19 00defd5c 7582356d stobject!ATL::CWindowImpl<CSysTray,ATL::CWindow,ATL::CWinTraits<2252341248,392> >::Create(struct HWND__ * hWndParent = 0x00000000, class ATL::_U_RECT rect = class ATL::_U_RECT, wchar_t * szWindowName = 0x00000000 "", unsigned long dwStyle = 0x86400000, unsigned long dwExStyle = 0x188, class ATL::_U_MENUorID MenuOrID = class ATL::_U_MENUorID, void * lpCreateParam = 0x00000000)+0xad [R:\src\dev\sdk\lib\atl\atlwin.h @ 1720] 
1a 00deffac 7582398b stobject!CSysTray::SysTrayThreadProc(void)+0x7d [R:\src\dev\dll\shellext\stobject\csystray.cpp @ 263] 
1b 00deffb8 7c64870d stobject!CSysTray::s_SysTrayThreadProc(void * param = 0x00149f70)+0x1b [R:\src\dev\dll\shellext\stobject\csystray.cpp @ 234] 
1c 00deffec 00000000 kernel32!BaseThreadStartup(<function> * lpStartAddress = 0x75823970, void * lpParameter = 0x00149f70)+0x5d [R:\src\dev\dll\win32\kernel32\client\thread.c @ 71] 

```